### PR TITLE
build: adding SLSA provenance generation for nupkg artifacts

### PR DESCRIPTION
**Pull request by rsoberano-ld**
**Requirements**

- [ ] I have added test coverage for new or changed functionality
- [ ] I have followed the repository's [pull request submission guidelines](../blob/main/CONTRIBUTING.md#submitting-pull-requests)
- [ ] I have validated my changes against all supported platform versions

**Related issues**

Provide links to any issues in this repository or elsewhere relating to this pull request.

**Describe the solution you've provided**

Provide a clear and concise description of what you expect to happen.

**Describe alternatives you've considered**

Provide a clear and concise description of any alternative solutions or features you've considered.

**Additional context**

Add any other context about the pull request here.

### CODE REVIEW

File: .github/workflows/manual-publish.yml [hunk 1]
```diff
@@ -45,3 +50,32 @@ jobs:
...
+      actions: read
+      id-token: write
+      contents: write
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@v1.10.0
+    if: ${{ inputs.pkg_name == 'LaunchDarkly.ServerSdk' }}
+    with:
+      base64-subjects: "${{ needs.build.outputs.server-sdk-hashes }}"
+      upload-assets: true
+      upload-tag-name: ${{ input.tag_name }}
+      provenance-name: ${{ format('LaunchDarkly.ServerSdk-{0}_provenance.intoto.jsonl', input.tag_name) }}
```
Comment (rsoberano-ld) [inline .github/workflows/manual-publish.yml:66]: How do we specify release tag name for manual publish workflows?

Comment (tanderson-ld) [reply .github/workflows/manual-publish.yml:66]: It is derived from the current version.  For example, the dotnet-server-sdk version is [here](https://github.com/launchdarkly/dotnet-server-sdk/blob/main/pkgs/sdk/server/src/LaunchDarkly.ServerSdk.csproj#L4).

File: .github/workflows/manual-publish.yml [hunk 2]
```diff
@@ -45,3 +50,32 @@ jobs:
...
+      actions: read
+      id-token: write
+      contents: write
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@v1.10.0
+    if: ${{ inputs.pkg_name == 'LaunchDarkly.ServerSdk.Telemetry' }}
+    with:
+      base64-subjects: "${{ needs.build.outputs.telemetry-hashes }}"
+      upload-assets: true
+      upload-tag-name: ${{ input.tag_name }}
+      provenance-name: ${{ format('LaunchDarkly.ServerSdk.Telemetry-{0}_provenance.intoto.jsonl', input.tag_name) }}
```
Comment (kinyoklion) [inline .github/workflows/manual-publish.yml:81]: Could use a newline here.

File: .github/workflows/manual-publish.yml (hunk 1 above)
Comment (kinyoklion) [reply .github/workflows/manual-publish.yml:66]: In C++ we made it so you could only publish an existing label. Which means you need to change the label if you are doing a manual publish to fix a problem. 

It is really inconvenient. Not sure if there would be a happy medium to handle both cases.

**Review by kinyoklion — APPROVED** [review]

**Pull request merged**